src/ParallelAlgorithms/Amr/Criteria/DriveToTarget.hpp

```diff
@@ -0,0 +1,124 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <pup.h>
+
+#include "Domain/Amr/Flag.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "ParallelAlgorithms/Amr/Criteria/Criterion.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace amr::Criteria {
+/*!
+ * \brief Refine the grid towards the target resolution and refinement levels in
+ * each dimension and then oscillate about the target.
+ *
+ * \details If the grid is at neither target in a given dimension, the
+ * flag chosen will be in the priority order Split,
+ * IncreaseResolution, DecreaseResolution, Join.
+ *
+ * \note To remain at the target, set the OscillationAtTarget Flags to
+ * DoNothing.
+ *
+ * \note This criterion is primarily for testing the mechanics of refinement.
+ */
+template <size_t Dim>
+class DriveToTarget : public Criterion {
+ public:
+  /// The target resolution
+  struct TargetResolution {
+    using type = std::array<size_t, Dim>;
+    static constexpr Options::String help = {
+        "The target resolution in each dimension."};
```

Comment:
Note here that "resolution" means number of grid points?

We've been a bit inconsistent throughout the code with this naming. Sometimes we use "number of grid points", sometimes it's "extents" (which isn't particularly clear because there's a notion of "size" attached to it), and in AMR it's "resolution" which also isn't super clear because h-refinement is also a notion of resolution. Nothing to resolve in this PR, but we should probably make this more consistent. I suggest "num points" because everyone understands what that means.